src/amo/components/SecondaryHero/index.js

```diff
@@ -2,19 +2,43 @@
 import invariant from 'invariant';
 import * as React from 'react';
 
-import type { SecondaryHeroShelfType } from 'amo/reducers/home';
+import { addParamsToHeroURL, isInternalURL } from 'amo/utils';
+import type {
+  HeroCallToActionType,
+  SecondaryHeroShelfType,
+} from 'amo/reducers/home';
 
 import './styles.scss';
 
+export const SECONDARY_HERO_SRC = 'homepage-secondary-hero';
+
 type Props = {| shelfData: SecondaryHeroShelfType |};
 
-export const SecondaryHeroBase = ({ shelfData }: Props) => {
+type InternalProps = {|
+  ...Props,
+  _isInternalURL: typeof isInternalURL,
+|};
+
+export const SecondaryHeroBase = ({
+  _isInternalURL = isInternalURL,
+  shelfData,
+}: InternalProps) => {
+  const makeCallToActionURL = (urlString: string) => {
```

Comment:
Can this function live outside the component?

Reply:
It would make it testable, if needed.

Otherwise, I find it a bit more readable. I usually put functions inside other functions when I need the context of the parent function in the inner function, which is not the case here.

It's not a change request, up to you :)

Reply:
Oh, I'm fine to do it, just wondering the rationale.